### PR TITLE
fix: ReplaceFullPath not working for root path (/)

### PR DESCRIPTION
**Pull request by eternalphane**
**What type of PR is this?**

**What this PR does / why we need it**:
`ReplaceFullPath` should support replacing root path during URL rewrite.

**Which issue(s) this PR fixes**:

### CODE REVIEW

**Review by zirain — CHANGES_REQUESTED** [review]
In case this happen, please kindly add an e2e test.

**Comment by arkodg** [comment]
@eternalphane can you share the config that not working ?
would be good to compare it with expectations from the spec https://gateway-api.sigs.k8s.io/reference/spec/#gateway.networking.k8s.io%2fv1.HTTPPathModifier

**Comment by eternalphane** [comment]
@arkodg [This simple example](https://github.com/kubernetes-sigs/gateway-api/blob/main/examples/standard/http-rewrite.yam) is not working when accessing the root path:
```yaml
apiVersion: gateway.networking.k8s.io/v1
kind: HTTPRoute
metadata:
  name: http-filter-1
  namespace: gateway-api-example-ns1
spec:
  parentRefs:
  - name: my-filter-gateway
    sectionName: http
  hostnames:
  - my-filter.example.com
  rules:
  - filters:
    - type: URLRewrite
      urlRewrite:
        path:
          type: ReplaceFullPath
          replaceFullPath: /foo
```

**Comment by arkodg** [comment]
@eternalphane is the request to my-filter.example.com  or my-filter.example.com/

**Comment by eternalphane** [comment]
> is the request to my-filter.example.com or my-filter.example.com/

Both seem to be the same 😂
image

File: internal/xds/translator/route.go [hunk 1]
```diff
@@ -314,7 +314,7 @@ func buildXdsURLRewriteAction(destName string, urlRewrite *ir.URLRewrite, pathMa
 		if urlRewrite.Path.FullReplace != nil {
 			routeAction.RegexRewrite = &matcherv3.RegexMatchAndSubstitute{
 				Pattern: &matcherv3.RegexMatcher{
-					Regex: "/.+",
+					Regex: "/.*",
```
Comment (arkodg) [inline internal/xds/translator/route.go:317]: hey sorry for the delay here, lets tighten this up a bit with `^/.*$`

so something random like `blah/` doesnt get accepted

**Comment by arkodg** [comment]
closed in favor of https://github.com/envoyproxy/gateway/pull/3530

**Pull request closed without merging**